### DISCUSSION TITLE
Share  ViewModel with Maui and BlazorWebView

### DISCUSSION THREAD

**Discussion by johngsandiford** [dotnet/maui#1530] [Q&A]
Hi. I have a MAUI app that mixes MAUI controls and BlazorWebViews.  I want to be able to share a view-model (i.e. the same instance) between both the MAUI part and the Blazor part,  but I can't figure out how to share it.

_Some view-models are one per SPA, but others are created dynamically._

Any help much appreciated.  Thanks, John

**Comment by DavidBerschauer** [comment]
I got the same problem. For pages with ViewModels registered as singletons, you could inject it into the Blazor component. Does anyone have a solution for transients?

**Comment by nor0x** [comment]
you can pass an existing ViewModel to the BlazorWebView. It can be retrieved via a CascadingParameter

more info:
https://johnnys.news/2024/03/Transient-ViewModels-with-NET-MAUI-Blazor-Hybrid
